Index: pages/upload.php
===================================================================
--- pages/upload.php	(revision 2079)
+++ pages/upload.php	(working copy)
@@ -509,7 +509,7 @@
 		fileMsg("<?php echo lang("_INVALID_FILE_EXT")." ".lang("_SELECT_ANOTHER_FILE") ?>");	
 		return false;
 		}
-   if (/^[^\\\/:;\*\?\"<>|]+(\.[^\\\/:;\*\?\"<>|]+)+$/.test(name)) 
+   if (/^[^\\\/:;\*\?\"<>|]+(\.[^\\\/:;\*\?\"<>|]+)*$/.test(name)) 
    {
 		return true; 
 	} else {
Index: classes/Functions.php
===================================================================
--- classes/Functions.php	(revision 2079)
+++ classes/Functions.php	(working copy)
@@ -692,7 +692,7 @@
 		// filename blank
 		if(isset($data["fileoriginalname"]) && $data["fileoriginalname"] === ""){ array_push($errorArray, "err_invalidfilename");}
 		// filename contains invalid characters
-		if(isset($data["fileoriginalname"]) && preg_match('=^[^\\\\/:;\*\?\"<>|]+(\.[^\\\\/:;\*\?\"<>|]+)+$=',$data["fileoriginalname"]) === 0){ array_push($errorArray, "err_invalidfilename");}
+		if(isset($data["fileoriginalname"]) && preg_match('=^[^\\\\/:;\*\?\"<>|]+(\.[^\\\\/:;\*\?\"<>|]+)*$=',$data["fileoriginalname"]) === 0){ array_push($errorArray, "err_invalidfilename");}
 		
 		// expiry out of range
 		if(strtotime($data["fileexpirydate"]) > strtotime("+".$config['default_daysvalid']." day") ||  strtotime($data["fileexpirydate"]) < strtotime("now"))
