Index: pages/vouchers.php
===================================================================
--- pages/vouchers.php	(revision 1089)
+++ pages/vouchers.php	(working copy)
@@ -36,7 +36,8 @@
  */
  ?>
 <script>
-var maximumDate= '<?php echo $config['default_daysvalid'] ?>';
+var maximumDate = <?php echo (time()+($config['default_daysvalid']*86400))*1000 ?>;
+var minimumDate = <?php echo (time()+86400)*1000 ?>;
 var maxEmailRecipients = <?php echo $config['max_email_recipients'] ?>;
 var datepickerDateFormat = '<?php echo lang('_DP_dateFormat'); ?>';
 var selectedVoucher = "";
@@ -47,9 +48,9 @@
 	
 	// stripe every second row in the tables
 	$("#vouchertable tr:odd").addClass('altcolor');
-	$("#datepicker" ).datepicker({ minDate: 1, maxDate: "+"+maximumDate+"D",altField: "#fileexpirydate", altFormat: "d-m-yy",currentText:maximumDate });
+	$("#datepicker" ).datepicker({ minDate: new Date(minimumDate), maxDate: new Date(maximumDate),altField: "#fileexpirydate", altFormat: "d-m-yy" });
 	$("#datepicker" ).datepicker( "option", "dateFormat", "<?php echo lang('_DP_dateFormat'); ?>" );
-	$("#datepicker").datepicker("setDate", new Date()+maximumDate);
+	$("#datepicker").datepicker("setDate", new Date(maximumDate));
 	
 	// set datepicker language
 	$.datepicker.setDefaults({
Index: pages/files.php
===================================================================
--- pages/files.php	(revision 1089)
+++ pages/files.php	(working copy)
@@ -75,15 +75,16 @@
 	
 	var selectedFile = ""; // file uid selected when deleteting
 	// set default maximum date for date datepicker
-	var maximumDate= '<?php echo $config['default_daysvalid']?>';
+	var maximumDate = <?php echo (time()+($config['default_daysvalid']*86400))*1000 ?>;
+	var minimumDate = <?php echo (time()+86400)*1000 ?>;
 	var maxEmailRecipients = <?php echo $config['max_email_recipients'] ?>;
 	var datepickerDateFormat = '<?php echo lang('_DP_dateFormat'); ?>';
 	
 	$(function() {
 		// initialise datepicker
-		$("#datepicker" ).datepicker({ minDate: 1, maxDate: "+"+maximumDate+"D",altField: "#fileexpirydate", altFormat: "d-m-yy" });
+		$("#datepicker" ).datepicker({ minDate: new Date(minimumDate), maxDate: new Date(maximumDate),altField: "#fileexpirydate", altFormat: "d-m-yy" });
 		$("#datepicker" ).datepicker( "option", "dateFormat", "<?php echo lang('_DP_dateFormat'); ?>" );
-		$("#datepicker").datepicker('setDate', new Date()+maximumDate);
+		$("#datepicker").datepicker("setDate", new Date(maximumDate));
 		
 		// set datepicker language
 		$.datepicker.setDefaults({
Index: pages/upload.php
===================================================================
--- pages/upload.php	(revision 1089)
+++ pages/upload.php	(working copy)
@@ -86,7 +87,8 @@
 	
 	// all default settings
 	var uploadid = '<?php echo $id ?>';
-	var maximumDate= '<?php echo $config['default_daysvalid']; ?>';
+	var maximumDate = <?php echo (time()+($config['default_daysvalid']*86400))*1000 ?>;
+	var minimumDate = <?php echo (time()+86400)*1000 ?>;
 	var maxHTML5uploadsize = <?php echo $config['max_html5_upload_size']; ?>;
 	var maxFLASHuploadsize = <?php echo $config['max_flash_upload_size']; ?>;
 	var maxEmailRecipients = <?php echo $config['max_email_recipients']; ?>;
@@ -109,9 +111,9 @@
 	$(function() { 
 
 		// set date picker
-		$("#datepicker" ).datepicker({ minDate: 1, maxDate: "+"+maximumDate+"D",altField: "#fileexpirydate", altFormat: "d-m-yy" });
+		$("#datepicker" ).datepicker({ minDate: new Date(minimumDate), maxDate: new Date(maximumDate),altField: "#fileexpirydate", altFormat: "d-m-yy" });
 		$("#datepicker" ).datepicker( "option", "dateFormat", "<?php echo lang("_DP_dateFormat"); ?>" );
-		$("#datepicker").datepicker("setDate", new Date()+maximumDate);
+		$("#datepicker").datepicker("setDate", new Date(maximumDate));
 			
 		// set datepicker language
 		$.datepicker.setDefaults({
